### PR TITLE
User guide restructure with folder hierarchy

### PR DESCRIPTION
**Pull request by aburdenthehand**
**What this PR does / why we need it**:

This is an alternative to #684 - mostly the same except it uses a folder hierarchy for the sections. This was after talking to some folks in the community that it is easier to navigate than one big bucket. It also has the added benefit of being able to demarcate ownership/responsibility of a section to a SIG.

The downside of this approach is that the redirects are more complicated (a global redirect could be used for the #684 approach). AFAIK this can be done vie the [mk docs redirects plugin](https://github.com/mkdocs/mkdocs-redirects) or in netlify itself. I would lean towards the former (and a minor update to the Dockerfile in project-infra). 

Edit: Oh yeah, while I was updating all the xrefs I made them all consistent (returning to root docs/ directory and including filepath) and converted some really old, dead docs urls to xrefs.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
Restructuring the user guide for better defined content
```

### CODE REVIEW

**Comment by aburdenthehand** [comment]
@jean-edouard @phoracek @alicefr @ksimon1 @machadovilaca @mhenriks @AlonaKaplan @0xFelix 
Can you please have a quick look at the relevant `.pages` files to ensure categorical accuracy. Thanks!

**Comment by aburdenthehand** [comment]
/hold
Until redirects are in place.

**Comment by jean-edouard** [comment]
> @jean-edouard @phoracek @alicefr @ksimon1 @machadovilaca @mhenriks @AlonaKaplan @0xFelix Can you please have a quick look at the relevant `.pages` files to ensure categorical accuracy. Thanks!

Looks good to me at first glance, I would just move live_migration.md to compute.
Thank you for doing this! The VM/Operations categories don't make sense, I always have to try both to find the page I'm looking for!

**Comment by aburdenthehand** [comment]
@jean-edouard Thanks for taking a look :) 
I've moved live_migration to compute as suggested.

File: docs/compute/.pages [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - node_assignment.md
+  - node_overcommit.md
+  - numa.md
+  - persistent_tpm_and_uefi_state.md
+  - resources_requests_and_limits.md
+  - run_strategies.md
+  - virtual_hardware.md
+  - memory_dump.md
+  - vsock.md
+  - windows_virtio_drivers.md
```
Comment (0xFelix) [inline docs/compute/.pages:19]: Might move that to user workloads?

**Review by 0xFelix — COMMENTED** [review]
One comment, but in general looks good to me.

**Review by 0xFelix — APPROVED** [review]
Thanks! To me it looks good.

/lgtm

**Comment by aburdenthehand** [comment]
/hold cancel
Redirect file now included

**Comment by phoracek** [comment]
/lgtm
/approve

A step in the right direction! Follow-up PRs (posted by sig members) should organize their sections

**Pull request merged**